### PR TITLE
Support handle eBPF Access Logs and update related UI

### PR DESCRIPTION
**Pull request by mrproliu**
Following https://github.com/apache/skywalking/discussions/11677, adding eBPF access log protocol handler and related UI Templates. 

- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [x] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

# Kubernetes Root Template

231705059230_ pic
image

# Kubernetes Service Relation

## TCP Related Metrics
151704604312_ pic

161704604329_ pic

## HTTP Related Metrics
171704604343_ pic

## Service Instance Topology

181704604386_ pic

# Service Instance Relation 

## TCP Related Metrics
191704604407_ pic

## HTTP Related Metrics
201704604417_ pic

# Endpoint

211704604442_ pic

# Self Obs Dashboards
251705278227_ pic

261705278239_ pic

### CODE REVIEW

**Comment by wu-sheng** [comment]
Some polish is required.
1. HTTP metrics and legends of normal and red nodes are required
2. OAL docs, sources especially, should be updated.
3. Observe Kubernetes doc should be updated accordingly.

File: docs/en/concepts-and-designs/scope-definitions.md [hunk 1]
```diff
@@ -414,3 +414,80 @@ This calculates the endpoint dimensional metrics data from each request of MQ sy
 | transmissionLatency | The latency from produce side to consume side .              |           | int(in ms) |
 | status              | Indicates the success or failure of the request.             |           | boolean    |
 | operation           | Indicates this access is on `Produce` or `Consume` side      |           | enum       |
+
+### SCOPES with `K8S` Prefix
+
+All metrics starting with `K8S` are derived from monitoring the Access Log in Kubernetes.
```
Comment (wu-sheng) [inline docs/en/concepts-and-designs/scope-definitions.md:420]:
```suggestion
All metrics starting with `K8S` are derived from Kubernetes monitoring by Rover(eBPF agent).
```

File: docs/en/concepts-and-designs/scope-definitions.md [hunk 2]
```diff
@@ -414,3 +414,80 @@ This calculates the endpoint dimensional metrics data from each request of MQ sy
...
+
+#### Service, Service Instance and relations
+
+For all `K8SService`, `K8SInstance`, `K8SServiceRelation` and `K8SServiceInstanceRelation`, they all have the
```
Comment (wu-sheng) [inline docs/en/concepts-and-designs/scope-definitions.md:424]: I think you missed those scopes' own field. Same for endpoint and endpoint relation.

File: docs/en/setup/backend/backend-k8s-monitoring-rover.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
+# Kubernetes (K8s) monitoring from Rover
+
+SkyWalking uses the SkyWalking Rover system to collect access logs from Kubernetes clusters and hands them over to the [OAL system](./../../concepts-and-designs/oal.md) for metrics and entity analysis.
+
+## Data flow
+1. SkyWalking Rover monitoring access log data from K8s and send to the OAP.
+2. The SkyWalking OAP Server receive access log from Rover through gRPC, analysis the generate entity, and using [OAL](../../concepts-and-designs/oal.md) to generating metrics.
+
+## Setup
+1. Setup [Rover in the Kubernetes](https://skywalking.apache.org/docs/skywalking-rover/next/en/setup/examples/deploy/kubernetes/readme/) and enable [access log service](https://skywalking.apache.org/docs/skywalking-rover/next/en/setup/configuration/accesslog/).
```
Comment (wu-sheng) [inline docs/en/setup/backend/backend-k8s-monitoring-rover.md:10]:
```suggestion
1. Setup [Rover in the Kubernetes](https://skywalking.apache.org/docs/skywalking-rover/next/en/setup/examples/deploy/kubernetes/readme/) and enable [access log service](https://skywalking.apache.org/docs/skywalking-rover/next/en/setup/configuration/accesslog/)(enabled by default).
```

File: docs/en/setup/backend/backend-k8s-monitoring-rover.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+2. Setup eBPF receiver module by the following configuration.
+```yaml
+```yaml
```
Comment (wu-sheng) [inline docs/en/setup/backend/backend-k8s-monitoring-rover.md:13]:
```suggestion
```yaml
```

Misformated?

File: docs/menu.yml [hunk 5]
```diff
@@ -203,7 +203,11 @@ catalog:
           - name: "Kubernetes"
             catalog:
               - name: "Observe Kubernetes"
-                path: "/en/setup/backend/backend-k8s-monitoring"
```
Comment (wu-sheng) [inline docs/menu.yml:206]: `/en/setup/backend/backend-k8s-monitoring` is referenced by blogs and SkyWalking UI, it should not be changed.

We should build a summary introduction doc for this path. And cAdvisor and eBPF should be two documents more about details.

File: docs/menu.yml [hunk 6]
```diff
@@ -203,7 +203,11 @@ catalog:
...
+                catalog:
+                  - name: "kube-state-metrics + cAdvisor"
+                    path: "/en/setup/backend/backend-k8s-monitoring-metrics-cadvisor"
+                  - name: "Rover"
+                    path: "/en/setup/backend/backend-k8s-monitoring-rover"
```
Comment (wu-sheng) [inline docs/menu.yml:210]:
```suggestion
                  - name: "Observe Kubernetes"
                    path: "/en/setup/backend/backend-k8s-monitoring"
                  - name: "kube-state-metrics + cAdvisor"
                    path: "/en/setup/backend/backend-k8s-monitoring-metrics-cadvisor"
                  - name: "Rover"
                    path: "/en/setup/backend/backend-k8s-monitoring-rover"
```

Please setup the docs in this way.

File: oap-server/server-receiver-plugin/skywalking-ebpf-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/ebpf/provider/EBPFReceiverProvider.java [hunk 7]
```diff
@@ -65,12 +67,19 @@ public void prepare() throws ServiceNotProvidedException, ModuleStartException {
...
+            .getService(OALEngineLoaderService.class)
+            .load(EBPFOALDefine.INSTANCE);
+
         final GRPCHandlerRegister grpcHandlerRegister = getManager().find(SharingServerModule.NAME)
                 .provider()
                 .getService(GRPCHandlerRegister.class);
         grpcHandlerRegister.addHandler(new EBPFProcessServiceHandler(getManager()));
         grpcHandlerRegister.addHandler(new EBPFProfilingServiceHandler(getManager()));
         grpcHandlerRegister.addHandler(new ContinuousProfilingServiceHandler(getManager(), this.config));
+        grpcHandlerRegister.addHandler(new AccessLogServiceHandler(getManager()));
```
Comment (wu-sheng) [inline oap-server/server-receiver-plugin/skywalking-ebpf-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/ebpf/provider/EBPFReceiverProvider.java:82]: Please refer to https://github.com/apache/skywalking/pull/11749, what we learned from that performance benchmark is that, for heavy traffic service, especially like ALS, we should have an option to provide a separate port to help traffic limitation and control.

File: oap-server/server-receiver-plugin/skywalking-ebpf-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/ebpf/provider/handler/AccessLogServiceHandler.java [hunk 8]
```diff
@@ -0,0 +1,700 @@
...
+            .getService(MetricsCreator.class);
+        this.inCounter = metricsCreator.createCounter(
+            "k8s_als_in_count", "The count of envoy ALS message received", MetricsTag.EMPTY_KEY,
+            MetricsTag.EMPTY_VALUE);
+        this.processHistogram = metricsCreator.createHistogramMetric(
+            "k8s_als_in_latency", "The process latency of envoy ALS message received", MetricsTag.EMPTY_KEY,
+            MetricsTag.EMPTY_VALUE);
+        this.dropCounter = metricsCreator.createCounter(
+            "k8s_als_drop_count", "The count of envoy ALS message dropped", MetricsTag.EMPTY_KEY,
+            MetricsTag.EMPTY_VALUE);
```
Comment (wu-sheng) [inline oap-server/server-receiver-plugin/skywalking-ebpf-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/ebpf/provider/handler/AccessLogServiceHandler.java:105]: As these important self-observability metrics, please update the native self-observability dashboard and Grafana dashboard, we should make them side by side with Envoy ALS and Agent traces metrics.

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/source/DefaultScopeDefine.java [hunk 9]
```diff
@@ -129,6 +129,14 @@ public class DefaultScopeDefine {
 
     public static final int SERVICE_HIERARCHY_RELATION = 70;
     public static final int INSTANCE_HIERARCHY_RELATION = 71;
+
+    public static final int K8S_SERVICE = 72;
+    public static final int K8S_SERVICE_INSTANCE = 73;
+    public static final int K8S_SERVICE_RELATION = 74;
+    public static final int K8S_SERVICE_INSTANCE_RELATION = 75;
+    public static final int K8S_ENDPOINT = 76;
+    public static final int K8S_ENDPOINT_REALATION = 77;
```
Comment (wu-sheng) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/source/DefaultScopeDefine.java:138]:
```suggestion
    // Rover(eBPF agent) for k8s monitoring
    public static final int K8S_SERVICE = 72;
    public static final int K8S_SERVICE_INSTANCE = 73;
    public static final int K8S_SERVICE_RELATION = 74;
    public static final int K8S_SERVICE_INSTANCE_RELATION = 75;
    public static final int K8S_ENDPOINT = 76;
    public static final int K8S_ENDPOINT_REALATION = 77;
```

File: docs/en/concepts-and-designs/scope-definitions.md (hunk 2 above)
Comment (mrproliu) [reply docs/en/concepts-and-designs/scope-definitions.md]: Updated.

File: docs/menu.yml (hunk 5 above)
Comment (mrproliu) [reply docs/menu.yml:206]: I have created a new summary document and linked these two documents together.

File: oap-server/server-receiver-plugin/skywalking-ebpf-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/ebpf/provider/handler/AccessLogServiceHandler.java (hunk 8 above)
Comment (mrproliu) [reply oap-server/server-receiver-plugin/skywalking-ebpf-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/ebpf/provider/handler/AccessLogServiceHandler.java:105]: Update, please take a look at the screenshots.

File: oap-server/server-receiver-plugin/skywalking-ebpf-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/ebpf/provider/EBPFReceiverProvider.java (hunk 7 above)
Comment (mrproliu) [reply oap-server/server-receiver-plugin/skywalking-ebpf-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/ebpf/provider/EBPFReceiverProvider.java]: Updated.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**